tests/main/views/test_outcomes.py

```diff
@@ -670,6 +670,7 @@ class TestUpdateOutcome(BaseApplicationTest, FixtureMixin):
                 ),
             ),
         )),
+        # help pytest make its printed representation of the parameter set a little more readable
         ids=(lambda val: "EMPTYDCT" if val == {} else None),
```

Comment:
Pointing out the typo seems churlish but now that this is in three places, I wonder if it could done via ` __str__` on `AnySupersetOf`?

Reply:
If you're talking about `DCT` I'm ... just ... trying ... to make it ... ever ... so ... much ... shorter... (the full parameter string is quite long)

Using the `str` value would defeat that and would pytest would probably reject it as it has non-alphanumeric chars.

Reply:
I'm still confused about what `ids` refers to or is used by. I'm suggesting you override `__str__` and hoping that pytest might use that, I don't see how it's in its gift to complain about what characters are in there.

Reply:
e.g. when you run `py.test -vv` it will end up listing the test variant as something like `test_direct_award_nonawarded_scenarios[none-suitable-True-True-True-cancelled-True-410-expected_response_data7]` to disambiguate it. It tries to limit the character set used because the idea is that this is sometimes taken as a command line parameter ("run just this test") and it wants it to be as un-shell-hostile as possible. Hence no curly braces. This function allows it to switch the opaque `expected_response_data7` for `EMPTYDCT` in cases where it's... an empty dict.